Simplenote/Classes/SPSidebarContainerViewController.m

```diff
@@ -1,421 +1,461 @@
+static const CGFloat SPSidebarAnimationDuration             = 0.4;
+static const CGFloat SPSidebarAnimationDamping              = 1.5;
+static const CGVector SPSidebarAnimationInitialVelocity     = {6, 0};
+static const CGFloat SPSidebarAnimationCompletionMin        = 0.001;
+static const CGFloat SPSidebarAnimationCompletionMax        = 0.999;
+static const CGFloat SPSidebarAnimationCompletionFactorFull = 1.0;
+static const CGFloat SPSidebarAnimationCompletionFactorZero = 0.0;
 
-@interface SPSidebarContainerViewController ()
+@interface SPSidebarContainerViewController () <UIGestureRecognizerDelegate>
 
-@property (nonatomic) CGPoint rootViewStartingOrigin;
-@property (nonatomic) CGPoint sidePanelStartingOrigin;
+@property (nonatomic, strong) UIViewController              *sidebarViewController;
+@property (nonatomic, strong) UIViewController              *mainViewController;
+@property (nonatomic, strong) UIViewPropertyAnimator        *animator;
+@property (nonatomic, strong) UITapGestureRecognizer        *mainViewTapGestureRecognier;
+@property (nonatomic, strong) UIPanGestureRecognizer        *panGestureRecognizer;
+@property (nonatomic, assign) BOOL                          isSidebarVisible;
+@property (nonatomic, assign) BOOL                          isPanningActive;
 
 @end
 
 @implementation SPSidebarContainerViewController
 
-- (instancetype)initWithSidebarViewController:(SPSidebarViewController *)sidebarViewController {
-    
+- (instancetype)initWithMainViewController:(UIViewController *)mainViewController
+                     sidebarViewController:(UIViewController *)sidebarViewController
+{
+    NSParameterAssert(mainViewController);
+    NSParameterAssert(sidebarViewController);
+
     self = [super init];
     if (self) {
-        
-        self.view.backgroundColor = [UIColor colorWithName:UIColorNameBackgroundColor];
-        
-        // add root view to screen
-        self.rootView = [[UIView alloc] initWithFrame:self.view.bounds];
-        self.rootView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
-        [self.view addSubview:_rootView];
-        
-        // setup gesture recognizers
-        self.panGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(viewDidPan:)];
-        self.panGestureRecognizer.delegate = self;
-        [self.rootView addGestureRecognizer:_panGestureRecognizer];
-        
-        _sidePanelViewController = sidebarViewController;
-        _sidePanelViewController.containerViewController = self;
+        self.mainViewController = mainViewController;
+        self.sidebarViewController = sidebarViewController;
+        self.automaticallyMatchSidebarInsetsWithMainInsets = YES;
+
+        [self configureMainView];
+        [self configurePanGestureRecognizer];
+        [self configureTapGestureRecognizer];
+        [self configureViewControllerContainment];
+        [self attachMainView];
+        [self attachSidebarView];
```

Comment:
Seems like most of this could be moved to `viewDidLoad` ?

Reply:
Definitely good call!!